src/ucp/core/ucp_context.h

```diff
@@ -267,6 +267,10 @@ typedef struct ucp_context {
     ucp_tl_md_t                   *tl_mds;    /* Memory domain resources */
     ucp_md_index_t                num_mds;    /* Number of memory domains */
 
+    /* Index of memory domain that have valid memory key for host memory
+       allocated with ucp_memh_alloc(). */
+    ucp_md_index_t                alloc_md_index;
```

Comment:
1. maybe rename to alloc_md_index or make it an array by mem type (and init only HOST entry)
2. comment "have valid memory key" seems wrong - all mds in reg_md_map have valid keys..